index.js

```diff
@@ -1,16 +1,50 @@
 'use strict';
-module.exports = function (hex) {
-	if (typeof hex !== 'string') {
-		throw new TypeError('Expected a string');
+
+const hexChars = 'a-f\\d';
+const match3or4Hex = `#?[${hexChars}]{3}[${hexChars}]?`;
+const match6or8Hex = `#?[${hexChars}]{6}([${hexChars}]{2})?`;
+
+const nonHexChars = new RegExp(`[^#${hexChars}]`, 'gi');
+const validHexSize = new RegExp(`^${match3or4Hex}$|^${match6or8Hex}$`, 'i');
+
+module.exports = function (hex, options = {}) {
+	if (typeof hex !== 'string' || nonHexChars.test(hex) || !validHexSize.test(hex)) {
+		throw new TypeError('Expected a valid hex string');
 	}
 
 	hex = hex.replace(/^#/, '');
+	let alpha = 255;
+
+	if (hex.length === 8) {
+		alpha = parseInt(hex.slice(6, 8), 16);
+		hex = hex.substring(0, 6);
+	}
+
+	if (hex.length === 4) {
+		alpha = parseInt(hex.slice(3, 4).repeat(2), 16);
+		hex = hex.substring(0, 3);
+	}
 
 	if (hex.length === 3) {
 		hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2];
 	}
 
-	var num = parseInt(hex, 16);
+	const num = parseInt(hex, 16);
+	const red = num >> 16;
+	const green = num >> 8 & 255; // eslint-disable-line no-mixed-operators
+	const blue = num & 255;
+	let result;
 
-	return [num >> 16, num >> 8 & 255, num & 255];
+	if (options.format === 'object') {
+		result = {red, green, blue};
+		if (alpha !== 255) {
+			result.alpha = alpha;
+		}
+	} else {
+		result = [red, green, blue];
+		if (alpha !== 255) {
```

Comment:
Alpha should always be included, for consistency.

Reply:
Do you think always including an alpha value would break existing use cases?

Reply:
It would, but this will have to be a major version anyways, as we're targeting a newer Node.js version, so it's fine. If anyone really needs the existing behavior, we can consider adding an option when/if they complain.